tests/unit/test_client.py

```diff
@@ -1902,7 +2113,14 @@ def test_set_iam_policy_invalid_policy(self):
         client = self._make_one(project=self.PROJECT, credentials=creds, _http=http)
 
         with self.assertRaises(TypeError):
-            client.set_iam_policy(self.TABLE_REF, invalid_policy_repr)
+            with mock.patch(
+                "google.cloud.bigquery.opentelemetry_tracing._get_final_span_attributes"
+            ) as final_attributes:
+                client.set_iam_policy(self.TABLE_REF, invalid_policy_repr)
+
+            final_attributes.assert_called_once_with(  # pragma: NO COVER
```

Comment:
Likewise, this line should be covered if outsized of the the `assertRaises` block.

```suggestion
        final_attributes.assert_called_once_with(
```